firmware/lucidgloves-firmware/_main.ino

```diff
@@ -42,6 +42,31 @@ void getInputs(void* parameter){
 }
 #endif
 
+#if (ENABLE_PCA_9865_SERVO && USING_FORCE_FEEDBACK)
+  #pragma once
+  #include "advancedConfig.h"
+  #include <Wire.h>
+  #include "PCA9685Servo.hpp"  
+#endif
+  
+struct {
+  void setup() {
+    #if (ENABLE_PCA_9865_SERVO && USING_FORCE_FEEDBACK)
+      Wire.begin(PIN_I2C_SDA, PIN_I2C_SCL);
+      Serial.println("I2C Initialized");
+    #endif
+  }
+} i2c;
+
+struct {
+  void setup() {
+    #if (ENABLE_PCA_9865_SERVO && USING_FORCE_FEEDBACK)
+      Initialize_PCA9685_Board();
+    #endif
+  }
+} PCA9685_Board;
```

Comment:
I think a lot of this could be abstracted into a servo class. Try see if you can follow the setup we have for communcation - there is an abstract class called `ICommunication` which each different communication method derives from. You could make a `Servo` abstract, class which both the ESP servo and `PCA9685` board derive from? That way you don't need to have this in the main file, and we can add support for more servos with in the future. Thanks.

Reply:
I see that there actually already was a function a few lines lower in _main.ino to setup the servos if using FFB. Since that function is created in haptics.ino I can drastically simplify things & no longer need to add anything to _main.ino if I simply change it from:
```
void setupServoHaptics(){
  pinkyServo.attach(PIN_PINKY_MOTOR);
  ringServo.attach(PIN_RING_MOTOR);
  middleServo.attach(PIN_MIDDLE_MOTOR);
  indexServo.attach(PIN_INDEX_MOTOR);
  thumbServo.attach(PIN_THUMB_MOTOR);
}
```
To:
```
struct {
  void setup() {
      Wire.begin(PIN_I2C_SDA, PIN_I2C_SCL);
      Serial.println("I2C Initialized");
  }
} i2c;

void setupServoHaptics(){
  #if (ENABLE_PCA_9865_SERVO)
    i2c.setup();
    Initialize_PCA9685_Board();
  #endif  
  pinkyServo.attach(PIN_PINKY_MOTOR);
  ringServo.attach(PIN_RING_MOTOR);
  middleServo.attach(PIN_MIDDLE_MOTOR);
  indexServo.attach(PIN_INDEX_MOTOR);
  thumbServo.attach(PIN_THUMB_MOTOR);
}
```

What are your thoughts on doing that instead of abstracting it into a servo class?

Reply:
I'd still probably prefer to have the different libraries in separate classes, that way you only need one `#if` for the whole class, instead of a lot of them scattered around. It would also make it more flexible if/when we needed to support any new servo libraries.

```c++
struct {
  void setup() {
      Wire.begin(PIN_I2C_SDA, PIN_I2C_SCL);
      Serial.println("I2C Initialized");
  }
} i2c;
```
I have no idea what doing `struct {} i2c` is actually called in c++ (unnamed struct?), though I would just make a function called `i2cSetup` or something as I'm not sure what having this in a struct really adds... though this will hopefully be moved into a servo class if you choose to split the servo libraries up into different classes.

Reply:
Sounds good. I'll still do the move to the preexisting function, squash that and some cleanup into my original commit. Then I'll do the move to an abstract Servo class as an additional commit as its a larger change to the codebase.

Reply:
Thanks! Feel free to reach out if you have any questions